codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Decoder.java

```diff
@@ -429,23 +404,113 @@ private static IllegalArgumentException notEnoughDataException(ByteBuf in) {
     }
 
     // Unsigned Little Endian Base 128 Variable-Length Integer Encoding
-    private static int decodeULE128(ByteBuf in) throws Http2Exception {
-        final int writerIndex = in.writerIndex();
-        for (int readerIndex = in.readerIndex(), shift = 0, result = 0;
-             readerIndex < writerIndex; ++readerIndex, shift += 7) {
-            byte b = in.getByte(readerIndex);
-            if (shift == 28 && (b & 0xF8) != 0) {
-                in.readerIndex(readerIndex + 1);
-                break;
-            }
-
-            if ((b & 0x80) == 0) {
+    private static int decodeULE128(ByteBuf in, int result) throws Http2Exception {
```

Comment:
Here is an updated by count when running the http2 unit test package (444 vs 140).

loop
```
5584 3115       3       io.netty.handler.codec.http2.internal.hpack.Decoder::decode (789 bytes)
                              @ 26   io.netty.buffer.ByteBuf::isReadable (0 bytes)   no static binding
                              @ 89   io.netty.buffer.ByteBuf::readByte (0 bytes)   no static binding
                              @ 171   io.netty.handler.codec.http2.internal.hpack.Decoder::indexHeader (80 bytes)   callee is too large
                              @ 242   io.netty.handler.codec.http2.internal.hpack.Decoder::readName (54 bytes)   callee is too large
                              @ 284   io.netty.handler.codec.http2.internal.hpack.Decoder::setDynamicTableSize (31 bytes)
                                @ 27   io.netty.handler.codec.http2.internal.hpack.DynamicTable::setCapacity (190 bytes)   callee is too large
                              @ 362   io.netty.handler.codec.http2.internal.hpack.Decoder::readName (54 bytes)   callee is too large
                              @ 377   io.netty.handler.codec.http2.internal.hpack.Decoder::decodeULE128 (140 bytes)   callee is too large
                              @ 380   io.netty.handler.codec.http2.internal.hpack.Decoder::setDynamicTableSize (31 bytes)
                                @ 27   io.netty.handler.codec.http2.internal.hpack.DynamicTable::setCapacity (190 bytes)   callee is too large
                              @ 392   io.netty.handler.codec.http2.internal.hpack.Decoder::decodeULE128 (140 bytes)   callee is too large
                              @ 398   io.netty.handler.codec.http2.internal.hpack.Decoder::indexHeader (80 bytes)   callee is too large
                              @ 412   io.netty.handler.codec.http2.internal.hpack.Decoder::decodeULE128 (140 bytes)   callee is too large
                              @ 415   io.netty.handler.codec.http2.internal.hpack.Decoder::readName (54 bytes)   callee is too large
```

manually unrolled
```
5465 3066       3       io.netty.handler.codec.http2.internal.hpack.Decoder::decode (789 bytes)
                              @ 26   io.netty.buffer.ByteBuf::isReadable (0 bytes)   no static binding
                              @ 89   io.netty.buffer.ByteBuf::readByte (0 bytes)   no static binding
                              @ 171   io.netty.handler.codec.http2.internal.hpack.Decoder::indexHeader (80 bytes)   callee is too large
                              @ 242   io.netty.handler.codec.http2.internal.hpack.Decoder::readName (54 bytes)   callee is too large
                              @ 284   io.netty.handler.codec.http2.internal.hpack.Decoder::setDynamicTableSize (31 bytes)
                                @ 27   io.netty.handler.codec.http2.internal.hpack.DynamicTable::setCapacity (190 bytes)   callee is too large
                              @ 362   io.netty.handler.codec.http2.internal.hpack.Decoder::readName (54 bytes)   callee is too large
                              @ 377   io.netty.handler.codec.http2.internal.hpack.Decoder::decodeULE128 (444 bytes)   callee is too large
                              @ 380   io.netty.handler.codec.http2.internal.hpack.Decoder::setDynamicTableSize (31 bytes)
                                @ 27   io.netty.handler.codec.http2.internal.hpack.DynamicTable::setCapacity (190 bytes)   callee is too large
                              @ 392   io.netty.handler.codec.http2.internal.hpack.Decoder::decodeULE128 (444 bytes)   callee is too large
                              @ 398   io.netty.handler.codec.http2.internal.hpack.Decoder::indexHeader (80 bytes)   callee is too large
                              @ 412   io.netty.handler.codec.http2.internal.hpack.Decoder::decodeULE128 (444 bytes)   callee is too large
                              @ 415   io.netty.handler.codec.http2.internal.hpack.Decoder::readName (54 bytes)   callee is too large
```